mitosheet/mitosheet/updates/replay_analysis.py

```diff
@@ -17,74 +17,48 @@
 REPLAY_ANALYSIS_UPDATE_EVENT = 'replay_analysis_update'
 REPLAY_ANALYSIS_UPDATE_PARAMS = [
     'analysis_name',
-    'import_summaries',
-    'clear_existing_analysis',
 ]
 
 def execute_replay_analysis_update(
         steps_manager: StepsManagerType,
         analysis_name: str,
-        import_summaries: Dict[str, Any],
-        clear_existing_analysis: bool
     ) -> None:
     """
     This function reapplies all the steps summarized in the passed step summaries, 
     which come from a saved analysis. 
 
-    If any of the step summaries fails, this function tries to roll back to before
-    it applied any of the stems
-
-    If clear_existing_analysis is set to true, then this will clear the entire widget
-    state container (except the initalize step) before applying the saved analysis.
+    If any of the step summaries fails, none of the analysis gets replayed at all.
     """
+    # When the frontend is refreshed, we read in and replay the anlaysis again
+    # unnecessarily, aka we need to read this
+    if analysis_name == steps_manager.analysis_name:
+        return
 
     # If we're getting an event telling us to update, we read in the steps from the file
     analysis = read_and_upgrade_analysis(analysis_name)
 
-    # If there is no analysis with this name, generate an error. This must occur before  
-    # steps_manager.steps = steps_manager.steps[:1] so that we don't clear all of the user's generated code. 
-    # This is particularly important for users who received the notebook from a colleague.
+    # If there is no analysis with this name, generate an error
     if analysis is None:
         log('replayed_nonexistant_analysis_failed')
```

Comment:
TODO: we need to add the log to this to the frontend, as it will no longer be triggered!